File: Array and String/Longest_common_prefix.cpp
class Solution {
public:    
    string longestCommonPrefix(vector<string>& strs) {
        int n = strs.size();
        if(n==0)
            return "";
        sort(strs.begin(),strs.end());
        string ans = "";
        for(int i=0;i<min(strs[0].length(),strs[n-1].length());i++)
        {
            if(strs[0][i]!=strs[n-1][i])
                break;
            ans+=strs[0][i];
        }
        return ans;
    }
};

File: Array and String/Longest_common_prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
        l = min(len(x) for x in strs)
        lcp = strs[0][:l]
        for i in range(1, len(strs)):
            while l >= 0 and lcp != strs[i][:l]:
                l -= 1
                lcp = strs[0][:l]
        return lcp
        